pilot/pkg/networking/plugin/authn/authentication.go

```diff
@@ -275,8 +275,12 @@ func buildSidecarListenerTLSContext(authenticationPolicy *authn.Policy) *auth.Do
 // OnOutboundListener is called whenever a new outbound listener is added to the LDS output for a given service
 // Can be used to add additional filters on the outbound path
 func (Plugin) OnOutboundListener(in *plugin.InputParams, mutable *plugin.MutableObjects) error {
-	// TODO: implementation
-	return nil
+	if in.Node.Type != model.Ingress && in.Node.Type != model.Router {
```

Comment:
If we only care about Ingress/Gateway, the if logic should be in.Node.Type == model.Ingress.

Why do we do this for model.Router?

Reply:
thanks, remove router.

Reply:
actually from integration test shows that node.Type for istio-ingressgateway is 'router'  now. It may be a bug or by design, for now I add 'router' type, and will check why ingress nodeType shows as 'router' later.